honeycombio/query_spec.go

```diff
@@ -0,0 +1,213 @@
+package honeycombio
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	honeycombio "github.com/kvrhdn/go-honeycombio"
+)
+
+var validQueryCalculationOps = []string{
+	"COUNT",
+	"SUM",
+	"AVG",
+	"COUNT_DISTINCT",
+	"MAX",
+	"MIN",
+	"P001",
+	"P01",
+	"P05",
+	"P10",
+	"P25",
+	"P50",
+	"P75",
+	"P90",
+	"P95",
+	"P99",
+	"P999",
+	"HEATMAP",
+}
+var validQueryFilterOps = []string{
+	"=",
+	"!=",
+	">",
+	">=",
+	"<",
+	"<=",
+	"starts-with",
+	"does-not-start-with",
+	"exists",
+	"does-not-exist",
+	"contains",
+	"does-not-contain",
+}
```

Comment:
Possible future TODO: Use the `go-honeycomb` enums?

https://github.com/kvrhdn/go-honeycombio/blob/eed63fb25297aadc235a18b2bef46388bd20ab7d/trigger.go#L67-L72

Reply:
I'm not super familiar with enums in Go. What would be the way to go? Terraform expects a `[]string`, so I guess I could create a list with all enums and map them with a utility func?
Or maybe I could add a function `isValidFilterOp(string) bool` to go-honeycombio and use that in the validator?